### DISCUSSION TITLE
Using dojo-webpack-plugin with `webpack serve`?

### DISCUSSION THREAD

**Discussion by ehuelsmann** [OpenNTF/dojo-webpack-plugin#358] [General]
Hi all, @chuckdumont,

As a first introduction, I'm the lead developer of an Open Source accounting and invoicing application (and more) called [LedgerSMB](https://github.com/ledgersmb/LedgerSMB). We've been using Dojo 1 for many years to give the application a professional look. In more recent times, we've been very thankfully using this plugin to optimize our JS delivery, lately even with Webpack5!

At the moment, we're at a point where we want to increase the use of JS in our project (moving more responsibilities to the client). In order to speed up our development speed, I'm trying `webpack serve` for hot reloading and immediate rebuilding of our JS while editing the source. However, I'm running into this error:

```plain
 > webpack --mode=development serve --progress --hot

76% sealing after chunk optimization dojo-webpack-plugin/srv/ledgersmb/node_modules/dojo-webpack-plugin/lib/DojoLoaderPlugin.js:269
					throw Error("Can't locate " + options.loader + " in compilation");
					^

Error: Can't locate /tmp/tmp-3775ajXHzH9kAONJ/dojo/dojo.js in compilation
    at /srv/ledgersmb/node_modules/dojo-webpack-plugin/lib/DojoLoaderPlugin.js:269:12
    at Set.forEach (<anonymous>)
    at DojoLoaderPlugin.afterOptimizeChunks (/srv/ledgersmb/node_modules/dojo-webpack-plugin/lib/DojoLoaderPlugin.js:266:10)
    at Hook.eval [as call] (eval at create (/srv/ledgersmb/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:12:1)
    at Hook.CALL_DELEGATE [as _call] (/srv/ledgersmb/node_modules/tapable/lib/Hook.js:14:14)
    at Compilation.seal (/srv/ledgersmb/node_modules/webpack/lib/Compilation.js:2957:34)
    at /srv/ledgersmb/node_modules/webpack/lib/Compiler.js:1187:20
    at /srv/ledgersmb/node_modules/webpack/lib/Compilation.js:2757:4
    at _next2 (eval at create (/srv/ledgersmb/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:23:1)
    at eval (eval at create (/srv/ledgersmb/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:58:1)
```
I've searched the documentation and didn't find anything which indicated that said that the plugin isn't compatible with `webpack serve` (or that modules even should comply with specific requirements). I'd love to debug this problem but the webpack (api) ecosystem is rather overwhelming and the plugin digs pretty deep into Dojo's internals. So before I do so, maybe you already know what the problem is or that the plugin isn't compatible with `webpack serve`? If it should be compatible, could you provide me a hint on where to look?

The state of my local repo is https://github.com/ehuelsmann/LedgerSMB/tree/11abc1d3185a0c51aa5d673e8d7da2aa7b32774b .

Thanks for your work on the plugin and thanks for any help or insight you can provide!

Regards,

Erik Huelsmann

**Comment by chuckdumont** [comment]
I suggest you try using a pre-built dojo loader. See [Pre-building the Dojo loader](https://github.com/OpenNTF/dojo-webpack-plugin#building-the-dojo-loader).

**Comment by ehuelsmann** [comment]
Thanks for the quick reply! I'll definitely try it.

One remark with respect to the question I asked: the file actually *does* exist when I check the path that's reported as failed. It seems like there's an ordering problem, as this does work when doing regular development and production builds.

Anyway, I'll report back when I have the pre-built dojo loader going.
